lib/ansible/modules/files/acl.py

```diff
@@ -201,7 +202,7 @@ def build_command(module, mode, path, follow, default, recursive, entry=''):
     if default:
         cmd.insert(1, '-d')
 
-    cmd.append(path)
+    cmd.append(shlex_quote(path))
```

Comment:
I think instead of quoting here we should instead pass the list of arguments into run_command on line 227.  The code would then look something like this:
``` python
 def run_acl(module, cmd, check_rc=True):
 
     try:
         (rc, out, err) = module.run_command('cmd, check_rc=check_rc)
     except Exception as e:
         module.fail_json(msg=to_native(e))
```

Reply:
That should have been:
``` python
         (rc, out, err) = module.run_command(cmd, check_rc=check_rc)
```

Reply:
In the python spirit, quoting seems to be the most explicit thing to do? 

" ".join(cmd) 

doesn't explain why escaping is needed. Quoting is also more secure.

Reply:
Passing a list of command srgs to run,_commsnd means that the Shell is never involved.  Instead each element of the list is a single argument.  That should be more secure than attempting to quote.